static_frame/core/frame.py

```diff
@@ -1393,6 +1393,59 @@ def from_json(cls,
                 consolidate_blocks=consolidate_blocks
                 )
 
+    
+    #@classmethod
+    #@doc_inject(selector='constructor_frame')
+    #def from_msgpack(cls,
+    #        msgpack_data: bin,
+    #        ) -> 'Frame':
+    @staticmethod
+    @doc_inject(selector='constructor_frame')
+    def from_msgpack(msgpack_data: bin):
+        '''Frame constructor from an in-memory binary object formatted as a msgpack.
+
+        Args:
+            msgpack_data: A binary msgpack object, encoding a object as produced from to_msgpack()
+
+        Returns:
+            :obj
+        '''
+        import msgpack
+        import msgpack_numpy
+        def decode(obj, chain=msgpack_numpy.decode):
+            if b'sf' in obj:
+                data = unpackb(obj[b'data']) #recursively call unpackb to deal with msgpacks within
+                clsname = obj[b'sf']
+                cls = globals()[clsname]
+                print('clsname', clsname)
+                if clsname in ['Frame']:
+                    return cls(
+                            TypeBlocks.from_blocks(data),
+                            name = obj[b'name'],
+                            index = unpackb(obj[b'index']),
+                            columns = unpackb(obj[b'columns']),
+                            own_data=True)
+                elif clsname in ['Index', 'IndexBase']:
+                    return cls(data, name = obj[b'name'])
+                elif clsname in ['IndexHierarchy', 'IndexHierarchyGO']:
+                    return cls(
+                            levels=unpackb(obj[b'levels']),
+                            name=obj[b'name'],
+                            blocks=TypeBlocks.from_blocks(data),
+                            own_blocks=True)
+                elif clsname in ['IndexLevel']:
+                    return cls(unpackb(obj[b'index']))
+            elif b'np' in obj:
+                data = msgpack.unpackb(obj[b'data'])
+                cls = getattr(np, obj[b'np'])
+                array = np.array([cls(d, unit) for d, unit in data])
```

Comment:
Rather than doing the list comprehension and applying `unit` to each item, if we store an `int` array and a single `unit`, we should be able to simply do `data.astype(unit)`, which will be very fast.

Reply:
I got it working like this:
`np.array(data, dtype=obj[b'dtype'])`
I think this should be the same performance right?